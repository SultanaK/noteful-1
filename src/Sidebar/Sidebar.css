.sidebar{
    min-width: 35%;
    background-color: #ffcdbf;
    min-height: 1000px;
}

.sidebar .active{
    background-color: lightcoral;
    margin: 0;
} 

.sidebar a, .sidebar p, .sidebar button{
    display: block;
    padding: 20px 0px 20px 20px;
    font-size: 1.4em;
    text-decoration: none;
}

.sidebar button{
    background-color: #ffcdbf;
    border: none;
    width: 100%;
    text-align: left;
}

.sidebar a:hover, .sidebar button:hover{
    background-color: lightcoral;
    margin: 0;
    transition: 0.6s;
    cursor: pointer;
}

svg{
    padding-right: 10px;
}

@media screen and (min-width: 800px){
    .sidebar{
        min-width: 300px;
    }
}
