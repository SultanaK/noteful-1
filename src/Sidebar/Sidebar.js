import React from 'react';
import PropTypes from 'prop-types';

import { NavLink } from 'react-router-dom';
import { faFolderOpen } from "@fortawesome/free-solid-svg-icons";
import { faFolderPlus } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import './Sidebar.css';

export default class Sidebar extends React.Component{
    constructor(props){
        super(props);
        this.handleFolderClick = this.handleFolderClick.bind(this);
        this.handleAddClick = this.handleAddClick.bind(this);
    }

    handleFolderClick(e){
        this.props.folderClickHandler(e.target.id);
    }

    handleAddClick(){
        this.props.addClickHandler();
    }

    render(){
        let folderDisplay = this.props.folders
            .map(folder => 
                <NavLink 
                    key={folder.folder_id} 
                    id={folder.folder_id}
                    onClick={this.handleFolderClick} 
                    aria-label="folder"                       
                    to={`/folders/${folder.folder_id}`}>
                        <FontAwesomeIcon icon={faFolderOpen}/>
                        {folder.folder_name}
                </NavLink>);
          return(
            <div className="sidebar">
                {folderDisplay}
                <button onClick={this.handleAddClick}>
                    <FontAwesomeIcon icon={faFolderPlus}/>
                    Add Folder
                </button>
            </div>
        );
    }
}

Sidebar.propTypes = {
    folders: PropTypes.array
};