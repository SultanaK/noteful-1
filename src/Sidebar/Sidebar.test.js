import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Sidebar from './Sidebar';

describe('Main sidebar', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(
        <BrowserRouter> 
            <Sidebar folders={sampleFolders}/>
        </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<BrowserRouter><Sidebar folders={sampleFolders}/></BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });

    //how to handle the onclick functions during testing? 
   /* it('loads dialog add folder window when clicked', () => {
      const wrapper = shallow(<Sidebar folders={sampleFolders} onClick={handleAddClick} />)
      wrapper.find('button').at(0).simulate('click');
   //   expect(toJson(wrapper)).toMatchSnapshot();
    })*/
});

const sampleFolders = [
  {
    "id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
    "name": "Important"
  },
  {
    "id": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
    "name": "Super"
  },
  {
    "id": "b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1",
    "name": "Spangley"
  }
];
