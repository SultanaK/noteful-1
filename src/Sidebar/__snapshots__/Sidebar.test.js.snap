// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main sidebar loads dialog add folder window when clicked 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Sidebar
    folders={
      Array [
        Object {
          "id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Important",
        },
        Object {
          "id": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Super",
        },
        Object {
          "id": "b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Spangley",
        },
      ]
    }
  />
</Router>
`;

exports[`Main sidebar renders the UI as expected 1`] = `
<div
  className="sidebar"
>
  <a
    aria-current={null}
    aria-label="folder"
    href="/folder/b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1"
    id="b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-folder-open fa-w-18 "
      data-icon="folder-open"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M572.694 292.093L500.27 416.248A63.997 63.997 0 0 1 444.989 448H45.025c-18.523 0-30.064-20.093-20.731-36.093l72.424-124.155A64 64 0 0 1 152 256h399.964c18.523 0 30.064 20.093 20.73 36.093zM152 224h328v-48c0-26.51-21.49-48-48-48H272l-64-64H48C21.49 64 0 85.49 0 112v278.046l69.077-118.418C86.214 242.25 117.989 224 152 224z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Important
  </a>
  <a
    aria-current={null}
    aria-label="folder"
    href="/folder/b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1"
    id="b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-folder-open fa-w-18 "
      data-icon="folder-open"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M572.694 292.093L500.27 416.248A63.997 63.997 0 0 1 444.989 448H45.025c-18.523 0-30.064-20.093-20.731-36.093l72.424-124.155A64 64 0 0 1 152 256h399.964c18.523 0 30.064 20.093 20.73 36.093zM152 224h328v-48c0-26.51-21.49-48-48-48H272l-64-64H48C21.49 64 0 85.49 0 112v278.046l69.077-118.418C86.214 242.25 117.989 224 152 224z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Super
  </a>
  <a
    aria-current={null}
    aria-label="folder"
    href="/folder/b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1"
    id="b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-folder-open fa-w-18 "
      data-icon="folder-open"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M572.694 292.093L500.27 416.248A63.997 63.997 0 0 1 444.989 448H45.025c-18.523 0-30.064-20.093-20.731-36.093l72.424-124.155A64 64 0 0 1 152 256h399.964c18.523 0 30.064 20.093 20.73 36.093zM152 224h328v-48c0-26.51-21.49-48-48-48H272l-64-64H48C21.49 64 0 85.49 0 112v278.046l69.077-118.418C86.214 242.25 117.989 224 152 224z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Spangley
  </a>
  <button
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-folder-plus fa-w-16 "
      data-icon="folder-plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M464 128H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V176c0-26.51-21.49-48-48-48zm-96 168c0 8.84-7.16 16-16 16h-72v72c0 8.84-7.16 16-16 16h-16c-8.84 0-16-7.16-16-16v-72h-72c-8.84 0-16-7.16-16-16v-16c0-8.84 7.16-16 16-16h72v-72c0-8.84 7.16-16 16-16h16c8.84 0 16 7.16 16 16v72h72c8.84 0 16 7.16 16 16v16z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Add Folder
  </button>
</div>
`;
